### ISSUE TITLE
`Process` should extend `AutoClosable`

### ISSUE THREAD

**Issue by 05nelsonm** [05nelsonm/kmp-process#32]
Should be able to utilized `Process` like a closable resource to ensure `Process.destroy` is always invoked.

```kotlin
val stdout = StringBuilder()

Process.Builder("ifconfig")
    .stdin(Stdio.Null)
    .stdout(Stdio.Pipe)
    .stderr(Stdio.Null)
    .spawn().use { p ->
        p.stdoutFeed { line ->
            if (stdout.isNotEmpty()) stdout.appendLine()
            stdout.append(line)
        }.waitfor()
    }

println(stdout.toString())
```

**Comment by 05nelsonm** [comment]
Adding the `blocked` label until https://youtrack.jetbrains.com/issue/KT-55777 is resolved (Kotlin 2.0)

**Comment by 05nelsonm** [comment]
I think that it's be best to implement this w/o extending `AutoClosable`, simply by adding an inline `spawn` function which takes in a lambda.

```kotlin
// Process.Builder.Companion

public companion object {

    @Throws(IOException::class)
    public inline fun <T: Any?> Process.Builder.spawn(block: () -> T): T {
        val p = spawn()

        val result = try {
            block(p)
        } finally {
            p.destroy()
        }

        return result
    }
}
```
